CMakeLists.txt

```diff
@@ -1,81 +1,150 @@
-set(serial_SRCS
-    src/serial.cc
-    include/serial/serial.h
-    include/serial/v8stdint.h
-)
+set(serial_SRCS src/serial.cc include/serial/serial.h)
+if (NOT HAVE_STDINT_H)
+    list(APPEND serial_SRCS include/serial/v8stdint.h)
+endif ()
+
 if(APPLE)
-    # If OSX
     list(APPEND serial_SRCS src/impl/unix.cc)
     list(APPEND serial_SRCS src/impl/list_ports/list_ports_osx.cc)
 elseif(UNIX)
-    # If unix
+    # linux
     list(APPEND serial_SRCS src/impl/unix.cc)
     list(APPEND serial_SRCS src/impl/list_ports/list_ports_linux.cc)
 else()
-    # If windows
+    # win32
     list(APPEND serial_SRCS src/impl/win.cc)
     list(APPEND serial_SRCS src/impl/list_ports/list_ports_win.cc)
 endif()
 
-## Add serial library
+set(serial_HEADERS include/serial/serial.h)
+if (NOT HAVE_STDINT_H)
+    list(APPEND serial_HEADERS include/serial/v8stdint.h)
+endif ()
+
+# Build  and link main library
 add_library(${PROJECT_NAME} ${serial_SRCS})
-if(APPLE)
-    target_link_libraries(${PROJECT_NAME} ${FOUNDATION_LIBRARY} ${IOKIT_LIBRARY})
-elseif(UNIX)
-    target_link_libraries(${PROJECT_NAME} rt pthread)
-else()
-    target_link_libraries(${PROJECT_NAME} setupapi)
-endif()
+set_target_properties(${PROJECT_NAME} PROPERTIES
+    VERSION ${PROJECT_VERSION}
+    SOVERSION ${PROJ_SOVERSION}
+    PUBLIC_HEADER "${serial_HEADERS}"
+)
+target_include_directories(${PROJECT_NAME} PUBLIC include)
+if (HAVE_STDINT_H)
+    target_compile_definitions(${PROJECT_NAME} PRIVATE -DHAVE_STDINT_H)
+endif ()
 
-## Uncomment for example
-add_executable(serial_example examples/serial_example.cc)
-add_dependencies(serial_example ${PROJECT_NAME})
-target_link_libraries(serial_example ${PROJECT_NAME})
+if (APPLE)
+    find_library(IOKIT_LIB IOKit)
+    find_library(FOUNDATION_LIB Foundation)
+    target_link_libraries(${PROJECT_NAME} ${FOUNDATION_LIB} ${IOKIT_LIB})
+elseif (UNIX)
+    target_link_libraries(${PROJECT_NAME} ${RT_LIBRARIES} ${PTHREAD_LIBRARIES})
+else ()
+    target_link_libraries(${PROJECT_NAME} setupapi)
+endif ()
 
-## Include headers
-include_directories(include)
 
-## Install executable
+## Install main library, possibly the catkin one.
 install(TARGETS ${PROJECT_NAME}
-    ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-    LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-    RUNTIME DESTINATION ${CATKIN_GLOBAL_BIN_DESTINATION}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+    PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}
 )
 
-## Install headers
-install(FILES include/serial/serial.h include/serial/v8stdint.h
-  DESTINATION ${CATKIN_GLOBAL_INCLUDE_DESTINATION}/serial)
-
-## Tests
-if(CATKIN_ENABLE_TESTING)
+# Other targets: test program, pkg-config and tests.
+if (CATKIN_ENABLE_TESTING)
+    include(CTest)
+    find_package(GTest REQUIRED)
+    enable_testing()
     add_subdirectory(tests)
 endif()
+
+if (DISABLE_CATKIN)
```

Comment:
Not sure why this has been if'ed on CATKINS

Reply:
So it can be built without the catkins dependency.